omegaconf/utils.py

```diff
@@ -0,0 +1,184 @@
+import re
+import six
+
+from .nodes import (
+    StringNode,
+    IntegerNode,
+    BooleanNode,
+    FloatNode,
+    UntypedNode,
+)
+
+if six.PY3:
+    from enum import Enum
+    from .nodes import EnumNode
+
+
+def node_wrap(type_, value):
+    from typing import Any
+
+    if type_ == Any:
+        node = UntypedNode(value)
+    elif isinstance(type_, type) and issubclass(type_, Enum):
+        node = EnumNode(type_)
+        node.set_value(value)
+    elif type_ == int:
+        node = IntegerNode(value)
+    elif type_ == float:
+        node = FloatNode(value)
+    elif type_ == bool:
+        node = BooleanNode(value)
+    elif type_ == str:
+        node = StringNode(value)
+    else:
+        raise ValueError("Unexpected object type : {}".format(type_.__name__))
+    return node
+
+
+def get_attr_data(obj):
+    import attr
+    import typing
+
+    d = {}
+    is_type = isinstance(obj, type)
+    obj_type = obj if is_type else type(obj)
+    for name, attrib in attr.fields_dict(obj_type).items():
+        type_ = attrib.type
+        is_nested = is_attr_class(type_)
+        if not is_type:
+            value = getattr(obj, name)
+        else:
+            value = attrib.default
+            if value == attr.NOTHING:
+                if is_nested:
+                    value = type_
+                else:
+                    raise ValueError(
+                        "Missing default value for {}, to indicate default must be populated later use '???'".format(
+                            name
+                        )
+                    )
+        if is_nested and value in (None, "???"):
```

Comment:
I think hard coded `"???"` is undesirable. For example, `omegaconf.UNK` like `attr.NOTHING` or  `dataclasses.MISSING` is preferable.